### PR TITLE
arm: Add Aspeed AST2700 platform support

### PR DESCRIPTION
**Pull request by ChiaweiW**
Aspeed AST2700 is a quad-core SoC with ARM Cortex-A35 CPU integrated.
This patch adds the initial platform support for AST2700 to execute 64-bits OP-TEE on top of ARMv8 TrustZone features.

### CODE REVIEW

File: core/arch/arm/plat-aspeed/conf.mk [hunk 1]
```diff
@@ -1,5 +1,3 @@
-PLATFORM_FLAVOR ?= ast2600
-
```
Comment (jforissier) [inline core/arch/arm/plat-aspeed/conf.mk:2]: This needs to stay

File: core/arch/arm/plat-aspeed/platform_ast2700.c [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+{
+	serial8250_uart_init(&console_data, CONSOLE_UART_BASE,
+			     CONSOLE_UART_CLK_IN_HZ, CONSOLE_BAUDRATE);
+	register_serial_console(&console_data.chip);
+}
+
+void plat_primary_init_early(void)
+{
+	/* TODO: add protection on demands */
+}
```
Comment (jforissier) [inline core/arch/arm/plat-aspeed/platform_ast2700.c:43]: Remove this function

**Review by jforissier — COMMENTED** [review]
Hi @ChiaweiW, two small comments, then:
`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

Also it would be nice to add this platform flavor to MAINTAINERS.

Thanks!

**Comment by ChiaweiW** [comment]
Hi @jforissier,

Change summary:
 - the two suggestions are addressed.
 - Add AST2700 platform flavor to MAINTAINER
 - Acked-by tags added

Thanks!

File: core/arch/arm/plat-aspeed/conf.mk [hunk 3]
```diff
@@ -18,6 +18,30 @@ CFG_TZDRAM_START ?= 0x88000000
...
+
+$(call force,CFG_8250_UART,y)
+$(call force,CFG_ARM64_core,y)
+$(call force,CFG_TEE_CORE_NB_CORE,4)
+$(call force,CFG_ARM_GICV3,y)
+$(call force,CFG_GIC,y)
+$(call force,CFG_WITH_ARM_TRUSTED_FW,y)
+$(call force,CFG_SECURE_TIME_SOURCE_CNTPCT,y)
+$(call force,CFG_CORE_ARM64_PA_BITS, 36)
+$(call force,CFG_LPAE_ADDR_SPACE_BITS, 36)
```
Comment (etienne-lms) [inline core/arch/arm/plat-aspeed/conf.mk:32]: can remove the space char here and at line above

File: core/arch/arm/plat-aspeed/platform_ast2700.c [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+#include <console.h>
+#include <drivers/gic.h>
+#include <drivers/serial8250_uart.h>
+#include <mm/core_memprot.h>
+#include <platform_config.h>
+#include <stdint.h>
+#include <io.h>
+#include <kernel/boot.h>
+#include <kernel/panic.h>
```
Comment (etienne-lms) [inline core/arch/arm/plat-aspeed/platform_ast2700.c:14]: Sort in alphabetical order.
_kernel/panic.h_, _io.h_ and _stdint.h_ not needed.
Replace _mm/core_memprot.h_ with _mm/core_mmu.h_.

**Comment by ChiaweiW** [comment]
Hi @etienne-lms,

Revised as suggested.
Thanks!

**Review by etienne-lms — COMMENTED** [review]
`Acked-by: Etienne Carriere <etienne.carriere@foss.st.com>`

**Comment by ChiaweiW** [comment]
> `Acked-by: Etienne Carriere <etienne.carriere@foss.st.com>`

Fix squashed and acked-by tags collected.
Thanks!

**Pull request merged**